### ISSUE TITLE
Running LLR

### ISSUE THREAD

**Issue by averyma** [yangarbiter/robust-local-lipschitz#1]
Great work!
I am hoping if it would be possible to share the config for LLR. (Similar to the last section of readme.)

Also, I noticed you have learning rate configurations for CURE regularization in one of the init files:

https://github.com/yangarbiter/robust-local-lipschitz/blob/45140271fb9993e5d28d8d6bba1151f42a4bc10a/lolip/models/__init__.py#L42

How was your experience implementing/reproducing CURE?

**Comment by yangarbiter** [comment]
The LLR setting can be found here

https://github.com/yangarbiter/robust-local-lipschitz/blob/45140271fb9993e5d28d8d6bba1151f42a4bc10a/lolip/models/torch_model.py#L200

we use is
```
lambd, mu = 4.0, 3.0
```

For CURE, we tried for a few weeks to implement it but it is not easy to train. In the end, we decided not to include this algorithm in our paper. One difference between our setting and the setting in their original paper is that they use a pretrained model. It is possible that it will be easier to train on a pretrained model.

**Comment by averyma** [comment]
Thanks for the reply!

I noticed the PGD20 accuracy in LLR paper is above 50%, while even the PGD10 accuracy reported in your paper is only around 20%. Do you know what might cause such a discrepancy?

**Comment by yangarbiter** [comment]
There are several difference between the experiment in the LLR's original paper and our experiment setting. But we are not sure which factor actually cause this discrepancy.
- They did not release their implementation, so our implementation may be different from their actual implementation
- They use ADAM to optimize the network (while we use SGD)
- They use softplus (not sure if they use it in hidden layer for just last layer), we use relu and softmax in the last layer
- They slowly increase the perturbation distance during training while we kept the number fixed throughout training
- They also add an L2 regularization to the network parameter while we didn't

Hope this helps.
